client/app/scripts/superdesk-templates/templates.js

```diff
@@ -156,8 +157,37 @@
                 desks.initialize()
                 .then(function() {
                     $scope.desks = desks.desks;
+
+                    /*
+                     * Returns desk name
+                     */
+                    $scope.getTemplateDesk = function getTemplateDesk(template) {
+                        return _.find($scope.desks._items , {_id: template.template_desk});
+                    };
+
+                    /*
+                     * Returns stage name
+                     */
+                    $scope.getTemplateStage = function getTemplateStage(template) {
+                        return _.find(desks.stages._items , {_id: template.template_stage});
+                    };
```

Comment:
Hey, good morning, you can sleep for a bit more, you know. ;)

Regarding the methods - in the promise success handler we simply define the methods, we don't invoke them. If the intent is to prevent the user from _using_ these methods before the data arrives (by no having the methods available yet), a better pattern IMO is to have some kind of a `hasLoaded` flag and disable the relevant UI elements based on it.

Would that do the trick, what do you think?